src/wallet/wallet.cpp

```diff
@@ -947,7 +963,10 @@ bool CWallet::AddToWalletIfInvolvingMe(const CTransactionRef& ptx, const CBlockI
             if (pIndex != nullptr)
                 wtx.SetMerkleBranch(pIndex, posInBlock);
 
-            return AddToWallet(wtx, false);
+            if (IsMine(tx) || IsFromMe(tx))
+                return AddToWallet(wtx, false);
+
+            return RemoveFromWallet(tx);
```

Comment:
Does it make sense to return the result of `RemoveFromWallet` in `AddToWalletIfInvolvingMe`?